src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1448,9 +1448,20 @@ trait Namers extends MethodSynthesis {
       }
 
       // Add a () parameter section if this overrides some method with () parameters
-      val vparamSymssOrEmptyParamsFromOverride =
-        if (overridden != NoSymbol && vparamSymss.isEmpty && overridden.alternatives.exists(_.info.isInstanceOf[MethodType])) ListOfNil // NOTE: must check `.info.isInstanceOf[MethodType]`, not `.isMethod`!
-        else vparamSymss
+      val vparamSymssOrEmptyParamsFromOverride = {
+        // must check `.info.isInstanceOf[MethodType]`, not `.isMethod`!
+        // Note that matching MethodType of NullaryMethodType must be nilary not nelary.
+        def overriddenNilary(sym: Symbol) = sym.info.isInstanceOf[MethodType]
+        if (overridden != NoSymbol && vparamSymss.isEmpty && overridden.alternatives.exists(overriddenNilary)) {
+          if (settings.warnNullaryOverride)
+            context.unit.toCheck += {() =>
+              def javaDetermined(sym: Symbol) = sym.isJavaDefined || isUniversalMember(sym)
+              if (settings.Ylint || !meth.overrides.exists(javaDetermined))
+                reporter.warning(ddef.pos, "nullary method assumes an empty parameter list from the overridden definition")
```

Comment:
Lose the lint, but the warning says, if the "add parens to your def" behavior is applied when no Java is involved, then maybe you didn't intend it. The warning was put under the other warning flag. I don't remember why it was necessary to defer to a post-typer `toCheck`. I guess forcing the overrides.